src/index.ts

```diff
@@ -39,14 +39,32 @@ const main = async () =>
     const percent = Math.round(data.percent * 100);
 
     //Log
-    info(`Downloaded ${transferred}K out of ${total}K (${percent}%)`);
+    if (!Number.isNaN(transferred) && !Number.isNaN(total) && !Number.isNaN(percent)) {
+      info(`Downloaded ${transferred}K out of ${total}K (${percent}%)`);
+    }
   };
-  await pipeline(
-    got.stream(isWindows ? 'https://win.wasmer.io' : 'https://get.wasmer.io').on('downloadProgress', progressHandler),
-    createWriteStream(tmp, {
-      mode: 0o655
-    })
-  );
+  const url = isWindows ? 'https://win.wasmer.io' : 'https://get.wasmer.io';
+  let retryAttempts = 0;
+  const maxRetryAttempts = 10;
+
+  while (retryAttempts < maxRetryAttempts) {
+    try {
+      await pipeline(
+        got.stream(url).on('downloadProgress', progressHandler),
+        createWriteStream(tmp, {
+          mode: 0o655
+        })
+      );
+      console.log('Downloaded installer.');
+      break;
+    }
+    catch (error) {
+      retryAttempts++;
+      if (retryAttempts >= maxRetryAttempts) {
+        throw new Error(`Failed to download installer after ${retryAttempts} attempts.\nLast error: ${error}`);
+      }
+    }
+  }
```

Comment:
Using pipelines + got + write-streams means I had to implement retry manually. It's not so bad though.

Reply:
There should be at least a small delay between attempts, like 5-10 seconds. 
(Ideally an increasing amount with exponential backoff, but a basic delay is allright)

Otherwise we could easily cause thundering herd problems for the backend after an outage.

Reply:
yeah, agreed on exponential backoffs.